SPEC.rdoc

```diff
@@ -140,18 +140,18 @@ The error stream. See the section below on the error stream for more information
 
 ==== <tt>rack.response_finished</tt>
 
-If present, an array of callables that will be run by the server after the response has been processed. The callables are called with <tt>environment, status, headers, error</tt> arguments and should not raise any exceptions. The callables would typically be called after sending the response to the client, but it could also be called if an error occurs while generating the response or sending the response (in that case, the +error+ argument will be a kind of +Exception+). The callables will be invoked in reverse order.
+If present, an array of callables that will be run by the server after the response has been processed. The callables are called with <tt>environment, status, headers, error</tt> arguments and should not raise any exceptions. The callables would typically be called after sending the response to the client, but it could also be called if an error occurs while generating the response or sending the response (in that case, the +error+ argument will be a kind of +Exception+). The callables will be called in reverse order.
 
 === The Input Stream
 
-The input stream is an +IO+-like object which contains the raw HTTP request data. When applicable, its external encoding must be "ASCII-8BIT" and it must be opened in binary mode. The input stream must respond to +gets+, +each+, and +read+:
-* +gets+ must be called without arguments and return a string, or +nil+ on EOF.
+The input stream is an +IO+-like object which contains the raw HTTP request data. When applicable, its external encoding must be <tt>ASCII-8BIT</tt> and it must be opened in binary mode. The input stream must respond to +gets+, +each+, and +read+:
+* +gets+ must be called without arguments and return a +String+, or +nil+ on <tt>EOF</tt>.
```

Comment:
I don't think we should use `<tt>` for EOF. EOF in this sense is a concept, not a specific identifier/code. Same for elsewhere in this file.

Reply:
```suggestion
* +gets+ must be called without arguments and return a +String+, or +nil+ on EOF (end-of-file).
```